lib/iris/tests/unit/experimental/ugrid/test_load_meshes.py

```diff
@@ -0,0 +1,218 @@
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Unit tests for the :func:`iris.experimental.ugrid.load_meshes` function.
+
+"""
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests  # isort:skip
+
+from pathlib import Path
+from shutil import rmtree
+from subprocess import check_call
+import tempfile
+from unittest import mock
+from uuid import uuid4
+
+from iris.experimental.ugrid import PARSE_UGRID_ON_LOAD, load_meshes, logger
+
+
+def setUpModule():
+    global TMP_DIR
+    TMP_DIR = Path(tempfile.mkdtemp())
+
+
+def tearDownModule():
+    if TMP_DIR is not None:
+        rmtree(TMP_DIR)
+
+
+def cdl_to_nc(cdl):
+    cdl_path = TMP_DIR / "tst.cdl"
+    nc_path = TMP_DIR / f"{uuid4()}.nc"
+    # Write CDL string into a temporary CDL file.
+    with open(cdl_path, "w") as f_out:
+        f_out.write(cdl)
+    # Use ncgen to convert this into an actual (temporary) netCDF file.
+    command = "ncgen -o {} {}".format(nc_path, cdl_path)
+    check_call(command, shell=True)
+    return str(nc_path)
+
+
+class TestsBasic(tests.IrisTest):
+    def setUp(self):
+        self.ref_cdl = """
+            netcdf mesh_test {
+                dimensions:
+                    node = 3 ;
+                    face = 1 ;
+                    vertex = 3 ;
+                    levels = 2 ;
+                variables:
+                    int mesh ;
+                        mesh:cf_role = "mesh_topology" ;
+                        mesh:topology_dimension = 2 ;
+                        mesh:node_coordinates = "node_x node_y" ;
+                        mesh:face_coordinates = "face_x face_y" ;
+                        mesh:face_node_connectivity = "face_nodes" ;
+                    float node_x(node) ;
+                        node_x:standard_name = "longitude" ;
+                    float node_y(node) ;
+                        node_y:standard_name = "latitude" ;
+                    float face_x(face) ;
+                        face_x:standard_name = "longitude" ;
+                    float face_y(face) ;
+                        face_y:standard_name = "latitude" ;
+                    int face_nodes(face, vertex) ;
+                        face_nodes:cf_role = "face_node_connectivity" ;
+                        face_nodes:start_index = 0 ;
+                    int levels(levels) ;
+                    float node_data(levels, node) ;
+                        node_data:coordinates = "node_x node_y" ;
+                        node_data:location = "node" ;
+                        node_data:mesh = "mesh" ;
+                    float face_data(levels, face) ;
+                        face_data:coordinates = "face_x face_y" ;
+                        face_data:location = "face" ;
+                        face_data:mesh = "mesh" ;
+                data:
```

Comment:
You probably don't need to define any actual variable data values to make this work?
Unless possibly some of the load chain checks for no-missing-points ?  : I haven't actually checked that

Reply:
I'd like to keep this. Tests are intimidating when revisiting, so it's important to me to minimise confusion wherever possible, and a good example is having valid CDL that one would actually expect to see.

Brevity isn't always good for clarity, even if it is in many cases!

Reply:
But it **_is_** valid CDL.  It just makes a file with all-missing variable content.

Reply:
OK "valid" is a loaded word. Would you ever expect to encounter such a file?

Reply:
Have discussed offline, leaving a minimal dataset in.